inst/stan/functions/obs_lpmf.stan

```diff
@@ -1,20 +1,32 @@
-real obs_lpmf(int[] dummy, int start, int end, int[,] obs, int[] sl,
+real obs_lpmf(int[] dummy, int start, int end, int[] obs, int[] sl, int[] csl,
               vector[] imp_obs, int[] sg, int[] st, int[,] rdlurd,
               vector srdlh, matrix ref_lh, int[] dpmfs, int ref_p, real phi) {
   real tar = 0;
   real tar_obs;
-  int t;
+  int init_cl = sl[start];
+  int start_n = csl[start] - sl[start];
+  int end_n = csl[end];
+  int n = end_n - start_n;
+  int snap_obs[n] = obs[(start_n + 1):end_n];
+  vector[n] exp_obs;
+  int start_snap, g, t, l;
+
   for (i in start:end) {
-    vector[sl[i]] exp_obs;
-    vector[sl[i]] rdlh;
+    g = sg[i];
+    t = st[i];
+    l = sl[i];
+    start_snap = csl[i] - start_n - l;
+    vector[l] exp_obs_by_rep;
+    vector[l] rdlh;
     // Find final observed/imputed expected observation
-    tar_obs = imp_obs[sg[i]][st[i]];
+    tar_obs = imp_obs[g][t];
     // allocate report day effects
-    rdlh = srdlh[rdlurd[st[i]:(st[i] + sl[i] - 1), sg[i]]];
+    rdlh = srdlh[rdlurd[t:(t + l - 1), g]];
```

Comment:
Like the new style with indices precomputed, makes it more readable!

Reply:
:)